lb/cross_reg_int_app_lb_mig_backend/main.tf

```diff
@@ -0,0 +1,384 @@
+
+# MIG_a
+# [START cloudloadbalancing_cross_inst_grp_mgr_a_tag]
+resource "google_compute_region_instance_group_manager" "mig_a" {
+  name     = "gl7-ilb-miga"
+  provider = google-beta
+  region   = "us-west1"
+  version {
+    instance_template = 
+google_compute_instance_template.instance_template_a.id
+    name              = "primary"
+  }
+  base_instance_name = "vm"
+  target_size        = 2
+}
+# [END cloudloadbalancing_cross_inst_grp_mgr_a_tag]
+
+# MIG_b
+# [START cloudloadbalancing_cross_inst_grp_mgr_b_tag]
+resource "google_compute_region_instance_group_manager" "mig_b" {
+  name     = "gl7-ilb-migb"
+  provider = google-beta
+  region   = "us-east1"
+  version {
+    instance_template = 
+google_compute_instance_template.instance_template_b.id
+    name              = "primary"
+  }
+  base_instance_name = "vm"
+  target_size        = 2
+}
+# [END cloudloadbalancing_cross_inst_grp_mgr_b_tag]
+
+# allow all access from health check ranges
+# [START cloudloadbalancing_cross_firewall_tag]
+resource "google_compute_firewall" "fw_healthcheck" {
+  name          = "gl7-ilb-fw-allow-hc"
+  provider      = google-beta
+  direction     = "INGRESS"
+  network       = google_compute_network.default.id
+  source_ranges = ["130.211.0.0/22", "35.191.0.0/16", "35.235.240.0/20"]
+  allow {
+    protocol = "tcp"
+  }
+}
+# [END cloudloadbalancing_cross_firewall_tag]
+
+# allow http and ssh traffic
+# [START cloudloadbalancing_cross_firewall_backend_tag]
+resource "google_compute_firewall" "fw_ilb_to_backends" {
+  name          = "fw-ilb-to-fw"
+  provider      = google-beta
+  network       = google_compute_network.default.id
+  source_ranges = ["0.0.0.0/0"]
+  allow {
+    protocol = "tcp"
+    ports    = ["22", "80", "443", "8080"]
+  }
+}
+# [END cloudloadbalancing_cross_firewall_backend_tag]
+
+# Allow http from proxy subnet to backends
+# [START cloudloadbalancing_cross_reg_firewall_proxy_tag]
+resource "google_compute_firewall" "fw_backends" {
+  name          = "gl7-ilb-fw-allow-ilb-to-backends"
+  direction     = "INGRESS"
+  network       = google_compute_network.default.id
+  source_ranges = ["10.129.0.0/23", "10.130.0.0/23"]
+  target_tags   = ["http-server"]
+  allow {
+    protocol = "tcp"
+    ports    = ["80", "443", "8080"]
+  }
+}
+# [END cloudloadbalancing_cross_reg_firewall_proxy_tag]
+
+# test instance
+# [START cloudloadbalancing_cross_vm_tst_tag]
+resource "google_compute_instance" "vm-test" {
+  name         = "test-vm"
+  provider     = google-beta
+  machine_type = "e2-small"
+  zone         = "us-west1-a"
+  network_interface {
+    network    = google_compute_network.default.id
+    subnetwork = google_compute_subnetwork.subnet_a.id
+    access_config {
+      # add external ip
+    }
+  }
+  boot_disk {
+    initialize_params {
+      image = "debian-cloud/debian-11"
+    }
+  }
+  # Without this, terraform thinks the image changes each time because
+  # bct-staging-images/debian-11 exands to
```

Comment:
Help me understand why this comment, when image `bct-staging-images/debian-11` is not being used here!

Reply:
This is a test-vm and the code was copied from a staging TF script. I have removed test-vm from the script.